crates/binstalk/src/fetchers/gh_crate_meta/hosting.rs

```diff
@@ -14,20 +15,63 @@ pub enum RepositoryHost {
 
 /// Make sure to update possible_dirs in `bins::infer_bin_dir_template`
 /// if you modified FULL_FILENAMES or NOVERSION_FILENAMES.
-pub const FULL_FILENAMES: &[&str] = &[
-    "{ name }-{ target }-v{ version }{ archive-suffix }",
-    "{ name }-{ target }-{ version }{ archive-suffix }",
-    "{ name }-{ version }-{ target }{ archive-suffix }",
-    "{ name }-v{ version }-{ target }{ archive-suffix }",
-    "{ name }_{ target }_v{ version }{ archive-suffix }",
-    "{ name }_{ target }_{ version }{ archive-suffix }",
-    "{ name }_{ version }_{ target }{ archive-suffix }",
-    "{ name }_v{ version }_{ target }{ archive-suffix }",
+pub const FULL_FILENAMES: &[Template<'_>] = &[
+    template!("/", { "name" }, "-", { "target" }, "-v", { "version" }, {
+        "archive-suffix"
+    }),
+    template!("/", { "name" }, "-", { "target" }, "-", { "version" }, {
+        "archive-suffix"
+    }),
+    template!("/", { "name" }, "-", { "version" }, "-", { "target" }, {
+        "archive-suffix"
+    }),
+    template!("/", { "name" }, "-v", { "version" }, "-", { "target" }, {
+        "archive-suffix"
+    }),
+    template!("/", { "name" }, "_", { "target" }, "_v", { "version" }, {
+        "archive-suffix"
+    }),
+    template!("/", { "name" }, "_", { "target" }, "_", { "version" }, {
+        "archive-suffix"
+    }),
+    template!("/", { "name" }, "_", { "version" }, "_", { "target" }, {
+        "archive-suffix"
+    }),
+    template!("/", { "name" }, "_v", { "version" }, "_", { "target" }, {
+        "archive-suffix"
+    }),
 ];
 
-pub const NOVERSION_FILENAMES: &[&str] = &[
-    "{ name }-{ target }{ archive-suffix }",
-    "{ name }_{ target }{ archive-suffix }",
+pub const NOVERSION_FILENAMES: &[Template<'_>] = &[
+    template!("/", { "name" }, "-", { "target" }, { "archive-suffix" }),
+    template!("/", { "name" }, "_", { "target" }, { "archive-suffix" }),
+];
+
+const GITHUB_RELEASE_PATHS: &[Template<'_>] = &[
+    template!({ "repo" }, "/releases/download/", { "version" }),
+    template!({ "repo" }, "/releases/download/v", { "version" }),
+];
+
+const GITLAB_RELEASE_PATHS: &[Template<'_>] = &[
+    template!(
+        { "repo" },
+        "/-/releases/",
+        { "version" },
+        "/downloads/binaries"
+    ),
+    template!(
+        { "repo" },
+        "/-/releases/v",
+        { "version" },
+        "/downloads/binaries"
+    ),
+];
+
+const BITBUCKET_RELEASE_PATHS: &[Template<'_>] = &[template!({ "repo" }, "/downloads")];
+
+const SOURCEFORGE_RELEASEW_PATHS: &[Template<'_>] = &[
```

Comment:
typo?

```suggestion
const SOURCEFORGE_RELEASE_PATHS: &[Template<'_>] = &[
```

Reply:
Yes, it's a typo.
Thanks for spotting that, I've fixed it.